### ISSUE TITLE
Misc feedback

### ISSUE THREAD

**Issue by tomayac** [human-centered-systems-lab/a11y-feedback#1]
- Some color contrast is insufficient.

   <img width="1663" alt="Screenshot 2024-05-14 at 11 39 56" src="https://github.com/human-centered-systems-lab/a11y-feedback/assets/145676/1b9569fd-65f6-4d02-ae28-d3cc4f2d79d3">

- Make use of [`autocomplete`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete). 
- Telephone input should use [`type=tel`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel).
- Data security notice [should not be part of the `label`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#accessibility_concerns).
- Consider checking the grammar with a native speaker of English.
- Create a GitHub Pages demo (or [StackBlitz preview](https://stackblitz.com/github/human-centered-systems-lab/a11y-feedback?file=README.md) or similar, so people can easily see the form.
